Client/Frontend/Menu/AppMenuConfiguration.swift

```diff
@@ -112,7 +116,11 @@ struct AppMenuConfiguration: MenuConfiguration {
             if #available(iOS 9, *) {
                 menuItems.append(AppMenuConfiguration.NewPrivateTabMenuItem)
             }
-            menuItems.append(tabState.isBookmarked ? AppMenuConfiguration.RemoveBookmarkMenuItem : AppMenuConfiguration.AddBookmarkMenuItem)
+            var menuItem = tabState.isBookmarked ? AppMenuConfiguration.RemoveBookmarkMenuItem : AppMenuConfiguration.AddBookmarkMenuItem
+            if let errorPageUrl = tabState.url where ErrorPageHelper.isErrorPageURL(errorPageUrl) {
```

Comment:
We should probably exclude more than just error pages. Reader mode pages or about pages, for example, shouldn't be bookmarkable either. So how about the following test:

`if let url = tabState.url where url.isWebPage(includeDataURIs: false) && !url.isLocal`